Source/WebKit/Shared/RTCPacketOptions.cpp

```diff
@@ -34,66 +34,35 @@
 
 namespace WebKit {
 
-void RTCPacketOptions::encode(IPC::Encoder& encoder) const
+RTCPacketOptions::RTCPacketOptions(const SerializableData& data)
 {
-    encoder << options.dscp;
-    encoder << safeCast<int32_t>(options.packet_id);
-    encoder << options.packet_time_params.rtp_sendtime_extension_id;
+    options.dscp = static_cast<rtc::DiffServCodePoint>(data.dscp);
+    options.packet_id = data.packetId;
 
-    encoder << static_cast<int64_t>(options.packet_time_params.srtp_auth_tag_len);
-    if (options.packet_time_params.srtp_auth_tag_len > 0)
-        encoder << IPC::DataReference(reinterpret_cast<const uint8_t*>(options.packet_time_params.srtp_auth_key.data()), options.packet_time_params.srtp_auth_key.size());
-
-    encoder << options.packet_time_params.srtp_packet_index;
-}
-
-std::optional<RTCPacketOptions> RTCPacketOptions::decode(IPC::Decoder& decoder)
-{
     rtc::PacketTimeUpdateParams params;
-    rtc::PacketOptions options;
-
-    rtc::DiffServCodePoint dscp;
-    if (!decoder.decode(dscp))
-        return std::nullopt;
-    options.dscp = dscp;
-
-    std::optional<int32_t> packetId;
-    decoder >> packetId;
-    if (!packetId)
-        return std::nullopt;
-    options.packet_id = packetId.value();
-
-    std::optional<int> rtpSendtimeExtensionId;
-    decoder >> rtpSendtimeExtensionId;
-    if (!rtpSendtimeExtensionId)
-        return std::nullopt;
-    params.rtp_sendtime_extension_id = rtpSendtimeExtensionId.value();
-
-    std::optional<int64_t> srtpAuthTagLength;
-    decoder >> srtpAuthTagLength;
-    if (!srtpAuthTagLength)
-        return std::nullopt;
-    params.srtp_auth_tag_len = srtpAuthTagLength.value();
-
-    if (params.srtp_auth_tag_len > 0) {
-        IPC::DataReference srtpAuthKey;
-        if (!decoder.decode(srtpAuthKey))
-            return std::nullopt;
-
-        params.srtp_auth_key = std::vector<char>(static_cast<size_t>(srtpAuthKey.size()));
-        memcpy(params.srtp_auth_key.data(), reinterpret_cast<const char*>(srtpAuthKey.data()), srtpAuthKey.size() * sizeof(char));
+    params.rtp_sendtime_extension_id = data.rtpSendtimeExtensionId;
+    params.srtp_auth_tag_len = data.srtpAuthTagLength;
+    if (data.srtpAuthTagLength > 0) {
+        auto* begin = reinterpret_cast<const char*>(data.srtpAuthKey.data());
+        params.srtp_auth_key = std::vector<char>(begin, begin + static_cast<size_t>(data.srtpAuthKey.size()));
```

Comment:
If we make srtpAuthKey a span<const char> we may be able to just use begin() and end() here.

Reply:
srtpAuthKey is a std::span. However, it contains `const uint8_t` instead of `char`.

Reply:
Fixed in the latest iteration. I am now using a std::span<const char> and using begin() / end() here.